pkg/apis/kubedirector/v1beta1/kubedirectorcluster_types.go

```diff
@@ -155,6 +156,14 @@ type ClusterStorage struct {
 	StorageClass *string `json:"storageClassName,omitempty"`
 }
 
+// BlockStorage defines the block storage type, path, and optionally size, if any, to be used
+// for mounting a block volume in a role.
+type BlockStorage struct {
+	StorageClass *string `json:"storageClassName,omitempty"`
+	Path         *string `json:"path,omitempty"`
+	Size         *string `json:"size,omitempty"`
```

Comment:
do we need size if we are always defaulting?

Reply:
@swamibluedata : Irrespective of what size we specify, the block size always remains the same. Any use case when this might be useful?

Reply:
block size inside the pod should reflect the size of the underlying block device?

Reply:
From what I found, a block volume occupies the entire partition/block of storage. Storage size in this context would be helpful when we check if the Claim Template Size is less than or equal to the storage capacity in PV.
eg: if PV is defined with storage capacity of 1Gi and the claim template requests 10Gi, the PV wont bind.

Reply:
I think we can keep it optional and modifiable by the user if, in case, the default value does not work.